winpkg/ddev_windows_installer.nsi

```diff
@@ -0,0 +1,1657 @@
+    Delete "$INSTDIR\mkcert uninstall.lnk"
+    Delete "$INSTDIR\ddev_uninstall.exe"
+
+    ; Remove icons and links directories
+    RMDir /r "$INSTDIR\Icons"
+    RMDir /r "$INSTDIR\Links"
+
+    ; Remove all installed shortcuts if they exist
+    !insertmacro MUI_STARTMENU_GETFOLDER "Application" $StartMenuGroup
+    ${If} "$StartMenuGroup" != ""
+        Delete "$SMPROGRAMS\$StartMenuGroup\DDEV.lnk"
+        Delete "$SMPROGRAMS\$StartMenuGroup\DDEV Website.lnk"
+        Delete "$SMPROGRAMS\$StartMenuGroup\DDEV Documentation.lnk"
+        Delete "$SMPROGRAMS\$StartMenuGroup\Uninstall ${PRODUCT_NAME}.lnk"
+        RMDir /r "$SMPROGRAMS\$StartMenuGroup\mkcert"
+        RMDir "$SMPROGRAMS\$StartMenuGroup"
+    ${EndIf}
+
+    ; Remove registry keys
+    DeleteRegKey ${REG_UNINST_ROOT} "${REG_UNINST_KEY}"
+    DeleteRegKey ${REG_INSTDIR_ROOT} "${REG_INSTDIR_KEY}"
+
+    ; Remove install directory if empty
+    RMDir "$INSTDIR"
+
+    ; Self-delete the uninstaller using ping approach
+    SetAutoClose true
+    ${If} ${FileExists} "$INSTDIR"
+        ExecWait 'cmd.exe /C ping 127.0.0.1 -n 2 && del /F /Q "$EXEPATH"'
+    ${EndIf}
+SectionEnd
+
+Function GetUbuntuDistros
+    DetailPrint "Starting GetUbuntuDistros..."
+    StrCpy $R0 ""  ; Result string
+
+    DetailPrint "Checking registry key..."
+    SetRegView 64
+    ClearErrors
+    EnumRegKey $R1 HKCU "Software\Microsoft\Windows\CurrentVersion\Lxss" 0
+    ${If} ${Errors}
+        DetailPrint "Error accessing Lxss registry key"
+        Push ""
+        Return
+    ${EndIf}
+    DetailPrint "Registry key exists and is accessible"
+
+    ; Count total number of keys first
+    StrCpy $R1 0   ; Index for enumeration
+    StrCpy $R5 0   ; Total count
+    count_loop:
+        ClearErrors
+        EnumRegKey $R2 HKCU "Software\Microsoft\Windows\CurrentVersion\Lxss" $R1
+        ${If} ${Errors}
+        ${OrIf} $R2 == ""
+            Goto count_done
+        ${EndIf}
+        IntOp $R5 $R5 + 1
+        IntOp $R1 $R1 + 1
+        Goto count_loop
+    count_done:
+    DetailPrint "Found $R5 total WSL distributions"
+
+    ; Now enumerate and check each key
+    StrCpy $R1 0   ; Reset index
+    ${While} $R1 < $R5
+        ClearErrors
+        EnumRegKey $R2 HKCU "Software\Microsoft\Windows\CurrentVersion\Lxss" $R1
+        ${If} ${Errors}
+            DetailPrint "Error enumerating key at index $R1"
+            Goto next_key
+        ${EndIf}
+
+        ClearErrors
+        ReadRegStr $R3 HKCU "Software\Microsoft\Windows\CurrentVersion\Lxss\$R2" "DistributionName"
+        ${If} ${Errors}
+            DetailPrint "Error reading DistributionName for key $R2"
+            Goto next_key
+        ${EndIf}
+
+        ; Check if it starts with "Ubuntu"
+        StrCpy $R4 $R3 6
+        ${If} $R4 == "Ubuntu"
+            DetailPrint "Found Ubuntu distribution: $R3"
+            ${If} $R0 != ""
+                StrCpy $R0 "$R0|"
+            ${EndIf}
+            StrCpy $R0 "$R0$R3"
+        ${EndIf}
```

Comment:
Let's check for `Flavor` instead of `DistributionName`:

![image](https://github.com/user-attachments/assets/47bb56f2-d6d5-4893-9c88-19e7c28af805)

Reply:
I like it! Requires exploring registry instead of wsl -l, but excellent

Reply:
Looks like we're already accessing the registry, aren't we? 🙂